### DISCUSSION TITLE
SO v2.3.21 - Change IP Address

### DISCUSSION THREAD

**Discussion by mgivens4182** [Security-Onion-Solutions/securityonion#2694] [Unsupported Versions] [answered]
I have been searching on how to change the IP address on the new 2.3.21 version and cannot find out how.  I am moving the SO server to the other side of the school and need to change the IP address for that network.

I really do not feel like rebuilding from scratch?

How is this not possible?

Please add this in a feature request.

Mike

**Comment by weslambert** [comment]
You can use `so-ip-update` to update the the IP address for a single-node install, but I can't guarantee that it will work for a distributed install, as there are plenty of moving parts -- manual updates may be required on the other nodes.

**Reply by mgivens4182** [reply]
Wes - thank you for the quick reply.  All I have is a stand alone so I am hoping this will cover it.  Is there a complete list of ALL the so-* commands?  I go back to the "cheat sheet" quite often - it would really help me being a beginner with SO.

Thanks again!  (will give that command a try when I get back to the school.)

MIke

**Reply by mgivens4182** [reply]
OK was really excited..  :-(  Did not work.  I tried the following:

sudo so-ip-update  (tried blank thinking it would ask me?)
then tried:
sudo so-ip-update X.X.X.X (tried adding the IP in there)
then tried:
sudo so-ip-update X.X.X.X/24  (tried adding the IP and sub)

ALL came back with the following error message:  (No matter what I put after so-ip-update)

"This tool will update a manager's IP address to the new IP assigned to the network management interface"
WARNING:  This tool is still undergoing  testing, use at your own risk!
Found old IP X.X.X.X
Detected new IP X.X.X.X  (Still the SAME IP)
ERROR: IP address has not changed
Exiting.

**Reply by TOoSmOotH** [reply]
I believe it has to be run after the IP has changed.

**Reply by weslambert** [reply]
> OK was really excited.. :-( Did not work. I tried the following:
> 
> sudo so-ip-update (tried blank thinking it would ask me?)
> then tried:
> sudo so-ip-update X.X.X.X (tried adding the IP in there)
> then tried:
> sudo so-ip-update X.X.X.X/24 (tried adding the IP and sub)
> 
> ALL came back with the following error message: (No matter what I put after so-ip-update)
> 
> "This tool will update a manager's IP address to the new IP assigned to the network management interface"
> WARNING: This tool is still undergoing testing, use at your own risk!
> Found old IP X.X.X.X
> Detected new IP X.X.X.X (Still the SAME IP)
> ERROR: IP address has not changed
> Exiting.

Yes, I should have clarified this should be run after the actual address has been changed for the interface.

**Comment by jertel** [comment]
The script was written expecting that the IP address had already changed on the interface. Are you running it while it's still on the old network? That would explain the output.

You can force it to use the new IP even if it's not yet assign to the interface via the syntax below:

```
NEW_IP=y.y.y.y so-ip-update
```

**Reply by mgivens4182** [reply]
So do I change the IP address the normal way on CentOS?  Go into: /etc/sysconfig/network-scripts/ifcfg-eth0 ???

Or is there a specific way with SO?  That is what I am trying to figure out?  

Then run the so-ip-update?

**Reply by TOoSmOotH** [reply]
Yes you can change it via ifcfg-eth0 and restart.

**Comment by mgivens4182** [comment, marked as answer]
OK was able to change the IP address by editing the config file.

Rebooted and all seems fine now.   Can get in via web console.

Thanks - Mike

**Reply by mgivens4182** [reply]
OK I now have officially given up on this idea.  Yes the IP did change, and I was able to get to the web interface.  However if I went to any other dashboards - the links within them still pointed to the old IP address.  I just do NOT have the confidence that this system is now correctly operating as the SO guys intended.   I have spent over 8 hours doing research and trying things on this - and have come to the conclusion that I should just rebuild and start fresh.  :-( 

Thanks to all who suggested how to go about this!

Mike

**Reply by maiwandh** [reply]
Unfortunately same problem here. I changed the config but I can´t access the web interface now. Neither on the old address or the new one. I just wanted to change my ip address in the same network. But seems not possible without complications.

**Reply by dougburks** [reply]
@maiwandh From https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Start a new discussion instead of replying to somebody else's discussion
Please search to see if you can find similar discussions that may help you. However, in order to avoid confusion, please do NOT reply to somebody else's discussion with your own issue. Instead, please start a new discussion and simply provide a hyperlink to the related discussion.
